frontend/lib/activation/deeplink_activation.dart

```diff
@@ -193,6 +202,11 @@ DynamicActivationCode? getActivationCode(BuildContext context, String base64qrco
   try {
     final activationCode = const ActivationCodeParser().parseQrCodeContent(const Base64Decoder().convert(base64qrcode));
     return activationCode;
+  } on CardExpiredException catch (e,_){
+    debugPrint('Die Karte ist abgelaufen: ${e.expiry}');
+    final expirationDate = DateTime.utc(e.expiry.year, e.expiry.month, e.expiry.day).millisecondsSinceEpoch ~/1000;
+    final activationCode = DynamicActivationCode(info: CardInfo(fullName: e.fullName,expirationDay:expirationDate));
+    return activationCode;
```

Comment:
i wouldn't return the activationCode on exception but null.
You can pass an `updateErrorMessage` function that updates on a particular exception the error message